### PR TITLE
Slpi dev

### PR DESCRIPTION
**Pull request by wangxdflight**

### CODE REVIEW

File: test/dspal_tester/adsp_proc/gpio_test_imp.c [hunk 1]
```diff
@@ -40,8 +40,9 @@
 #include "test_status.h"
 #include "test_utils.h"
 
-#define GPIO_DEVICE_PATH  "/dev/gpio-56"
-#define GPIO_INT_DEVICE_PATH  "/dev/gpio-55"
+#define GPIO_DEVICE_PATH  "/dev/gpio-10"
+#define GPIO_DEVICE_PATH_LOOPBACK  "/dev/gpio-11"
+#define GPIO_INT_DEVICE_PATH  "/dev/gpio-11"
```
Comment (mcharleb) [inline test/dspal_tester/adsp_proc/gpio_test_imp.c:46]: Lets move these to test/dspal_tester/adsp_proc/platform.h and localize ifdefs for ADSP and SLPI differences.

File: test/dspal_tester/adsp_proc/gpio_test_imp.c [hunk 2]
```diff
@@ -216,6 +217,100 @@ int dspal_tester_test_gpio_read_write(void)
 	return result;
 }
 
+/**
```
Comment (mcharleb) [inline test/dspal_tester/adsp_proc/gpio_test_imp.c:220]: The documentation for DSPAL tester (not just DOXYGEN docs) need to explain the HW setup needed to run this test.

File: test/dspal_tester/adsp_proc/gpio_test_imp.c [hunk 3]
```diff
@@ -246,10 +341,10 @@ void *gpio_int_isr(DSPAL_GPIO_INT_ISR_CTX context)
 * GPIO interrupt (the 2 pins should be wired together)
 
 * Test:
-* 1) Opens file for GPIO A device (IO Pin) (gpio-56)
+* 1) Opens file for GPIO A device (IO Pin) (gpio-10)
 * 2) Uses ioctl to set the GPIO pin A to me in IO mode and checks that this succeed
```
Comment (mcharleb) [inline test/dspal_tester/adsp_proc/gpio_test_imp.c:345]: Use a define like DSPAL_GPIO_A in platform.h

File: test/dspal_tester/adsp_proc/gpio_test_imp.c [hunk 4]
```diff
@@ -246,10 +341,10 @@ void *gpio_int_isr(DSPAL_GPIO_INT_ISR_CTX context)
...
 * 3) Sets the GPIO pin A to be LOW
-* 4) Opens file for GPIO B device (interrupt Pin) (gpio-55)
+* 4) Opens file for GPIO B device (interrupt Pin) (gpio-11)
 * 5) ioctl the pin B to be a hardware interrupt pin (triggered on a rising edge)
```
Comment (mcharleb) [inline test/dspal_tester/adsp_proc/gpio_test_imp.c:348]: Use a define like DSPAL_GPIO_B in platform.h

File: test/dspal_tester/adsp_proc/i2c_test_imp.c [hunk 5]
```diff
@@ -37,17 +37,23 @@
...
+#include "test_utils.h"
+
 /**
 * @brief Test to see i2c device can be opened and configured.
 *
 * @par
 * Test:
-* 1) Open the i2c device (/dev/i2c-8)
+* 1) Open the i2c device (/dev/i2c-3) 
 * 2) Configure the i2c device to have (using ioctl):
```
Comment (mcharleb) [inline test/dspal_tester/adsp_proc/i2c_test_imp.c:48]: Use a define like DSPAL_TEST_I2C_DEVNAME in platform.h

File: test/dspal_tester/adsp_proc/i2c_test_imp.c [hunk 6]
```diff
@@ -37,17 +37,23 @@
...
-*     -Slave address: 0x70
+*     -Slave address: 0x77
 *     -Bus Frequency in khz: 400
```
Comment (mcharleb) [inline test/dspal_tester/adsp_proc/i2c_test_imp.c:50]: Use a define like DSPAL_TEST_I2C_SLAVE_ADDR in platform.h

File: test/dspal_tester/adsp_proc/i2c_test_imp.c [hunk 7]
```diff
@@ -56,25 +62,55 @@
 int dspal_tester_i2c_test(void)
 {
 	int ret = SUCCESS;
+    uint8_t buf[2]; 
 	/*
 	 * Open i2c device
 	 */
 	int fd = -1;
-	fd = open("/dev/i2c-8", 0);
+	fd = open("/dev/i2c-2", 0);
```
Comment (mcharleb) [inline test/dspal_tester/adsp_proc/i2c_test_imp.c:71]: Use a define like DSPAL_TEST_I2C_DEVNAME in platform.h

File: test/dspal_tester/adsp_proc/i2c_test_imp.c [hunk 8]
```diff
@@ -56,25 +62,55 @@
...
 
 	if (fd > 0) {
 		/*
 		 * Configure I2C device
 		 */
 		struct dspal_i2c_ioctl_slave_config slave_config;
-		slave_config.slave_address = 0x70;
+		slave_config.slave_address = 0x76;
 		slave_config.bus_frequency_in_khz = 400;
```
Comment (mcharleb) [inline test/dspal_tester/adsp_proc/i2c_test_imp.c:78]: Use a define like DSPAL_TEST_I2C_SLAVE_ADDR in platform.h

File: test/dspal_tester/adsp_proc/i2c_test_imp.c [hunk 9]
```diff
@@ -56,25 +62,55 @@
...
 		slave_config.byte_transer_timeout_in_usecs = 9000;
 
 		if (ioctl(fd, I2C_IOCTL_CONFIG, &slave_config) != 0) {
 			ret = ERROR;
 		}
```
Comment (mcharleb) [inline test/dspal_tester/adsp_proc/i2c_test_imp.c:84]: Missing parenthesis?

File: test/dspal_tester/adsp_proc/posix_pthread_tests.c [hunk 10]
```diff
@@ -843,11 +843,22 @@ int dspal_tester_test_pthread_cond_timedwait(void)
 	int test_value = TEST_FAIL;
 	pthread_t thread;
 
+	pthread_attr_t attr;
+	size_t stacksize = 2 * 1024;
```
Comment (mcharleb) [inline test/dspal_tester/adsp_proc/posix_pthread_tests.c:847]: Add note that this is required for SLPI and still works on ADSP

File: test/dspal_tester/adsp_proc/posix_time_tests.c [hunk 11]
```diff
@@ -257,8 +265,108 @@ int dspal_tester_test_usleep(void)
...
+	uint64_t start_time = dspal_get_time_1us();
+	int counter = 5000;
+	for (int i = 0; i < counter; i ++) {		
+		uint64_t curr_time = dspal_get_time_1us();
+	}
+	int elapsed_time = dspal_get_time_1us() - start_time;
+	LOG_ERR("total %d dspal_get_time_1us calls takes %d, each %f", counter, elapsed_time, elapsed_time*1.0/counter);
+	
+}
+int test_usleep_worker(int flag)
```
Comment (mcharleb) [inline test/dspal_tester/adsp_proc/posix_time_tests.c:284]: Need to document expected latency, that threads are sched FIFO, and any thread at the same priority or higher can create latency.

File: test/dspal_tester/adsp_proc/serial_test_imp.c [hunk 12]
```diff
@@ -56,14 +58,22 @@
  * - make sure to define UART-BAM mappings in /usr/share/data/adsp/blsp.config
  *   See snapdragon flight user guide for details.
  */
-#define MAX_UART_DEVICE_NUM      6
-#define NUM_UART_DEVICE_ENABLED  4
+#define MAX_UART_DEVICE_NUM      1 //6
+#define NUM_UART_DEVICE_ENABLED  1 //4
```
Comment (mcharleb) [inline test/dspal_tester/adsp_proc/serial_test_imp.c:63]: Either this is "1" and the comment should be removed, or there is a TODO or FIXME not required here

File: test/dspal_tester/adsp_proc/serial_test_imp.c [hunk 13]
```diff
@@ -102,17 +112,184 @@ int dspal_tester_serial_multi_port_open(void)
...
+		LOG_ERR("/dev/tty-%d read callback received bytes: %d",
+			 rx_dev_id, num_bytes);
+
+        for (int i = 0; i < (int)num_bytes; i++) 
+        {
+            //LOG_ERR("buffer[%d]=%x", i , buffer[i]);
+        }
+		if (num_bytes != 12)
+			LOG_ERR("!!!!!!!!!!!!!!!!!!available bytes %d", num_bytes);
+
```
Comment (mcharleb) [inline test/dspal_tester/adsp_proc/serial_test_imp.c:160]: Please fix this error output

File: test/dspal_tester/adsp_proc/serial_test_imp.c [hunk 14]
```diff
@@ -102,17 +112,184 @@ int dspal_tester_serial_multi_port_open(void)
...
+            LOG_ERR("failed to write");
+            return NULL; 
+        }
+
+        usleep(10000); 
+		//usleep(1000*1000); //added for serial TX debug;
+    }
+
+#if 0 
+    /* Assign IDs:
```
Comment (mcharleb) [inline test/dspal_tester/adsp_proc/serial_test_imp.c:192]: Please remove if node not needed

File: test/dspal_tester/adsp_proc/serial_test_imp.c [hunk 15]
```diff
@@ -405,6 +582,138 @@ int dspal_tester_serial_read_with_small_buffer(void)
...
+			 (serial_i2c_fildes < SUCCESS) ? "fail" : "succeed");
+
+    if (serial_i2c_fildes < SUCCESS)  {
+        LOG_ERR("serial multi-port read/write callback test FAILED");
+        result = ERROR; 
+        return result; 
+    }
+
+#if 0 
+     /*Spin motor*/
```
Comment (mcharleb) [inline test/dspal_tester/adsp_proc/serial_test_imp.c:690]: Please remove if code not needed

File: test/dspal_tester/adsp_proc/serial_test_imp.c [hunk 16]
```diff
@@ -405,6 +582,138 @@ int dspal_tester_serial_read_with_small_buffer(void)
...
+    } else {
+        LOG_ERR("failed to write");
+        return NULL; 
+    }*/
+
+    usleep(10000); 
+
+    //usleep(SERIAL_WRITE_DELAY_IN_USECS);
+
+	close(serial_i2c_fildes);
```
Comment (mcharleb) [inline test/dspal_tester/adsp_proc/serial_test_imp.c:736]: Please format your code. There is inconsistent use of tabs and spaces

File: test/dspal_tester/adsp_proc/spi_test_imp.c [hunk 17]
```diff
@@ -41,7 +41,7 @@
 #include "test_status.h"
 #include "test_utils.h"
 
-#define SPI_DEVICE_PATH "/dev/spi-8"
+#define SPI_DEVICE_PATH "/dev/spi-1"
 #define SPI_TEST_CYCLES 10
```
Comment (mcharleb) [inline test/dspal_tester/adsp_proc/spi_test_imp.c:45]: Please move to platform.h with appropriate ifdefs

File: test/dspal_tester/apps_proc/dspal_tester_main.c [hunk 18]
```diff
@@ -44,6 +44,12 @@
 #include "posix_test_suite.h"
 #include "io_test_suite.h"
 
+#if 0
+#include "remote.h"
```
Comment (mcharleb) [inline test/dspal_tester/apps_proc/dspal_tester_main.c:48]: Remove if code not needed

File: test/dspal_tester/apps_proc/posix_test_suite.c [hunk 19]
```diff
@@ -38,9 +38,11 @@
 int run_posix_test_suite(void)
 {
 	int test_results = TEST_PASS;
+	LOG_INFO("testing malloc size limit");
+#if 0
+	test_results |= display_test_results(dspal_tester_test_malloc(), "malloc size test");
```
Comment (mcharleb) [inline test/dspal_tester/apps_proc/posix_test_suite.c:43]: Remove if code not needed

File: test/dspal_tester/apps_proc/posix_test_suite.c [hunk 20]
```diff
@@ -50,15 +52,20 @@ int run_posix_test_suite(void)
...
 	test_results |= display_test_results(dspal_tester_test_clock_gettime(), "clock_gettime");
 	test_results |= display_test_results(dspal_tester_test_clock_settime(), "clock_settime");
 	test_results |= display_test_results(dspal_tester_test_one_shot_timer_cb(), "one shot timer cb");
 	test_results |= display_test_results(dspal_tester_test_periodic_timer_cb(), "periodic timer cb");
 	test_results |= display_test_results(dspal_tester_test_periodic_timer_signal_cb(), "periodic timer signal cb");
 	test_results |= display_test_results(dspal_tester_test_periodic_timer_sigwait(), "periodic timer sigwait");
-
+#endif
+#if 0
 	LOG_INFO("testing pthread.h");
```
Comment (mcharleb) [inline test/dspal_tester/apps_proc/posix_test_suite.c:69]: remove if code not needed

File: test/dspal_tester/dspal_tester.idl [hunk 21]
```diff
@@ -37,7 +37,7 @@
 
 interface dspal_tester
 {
-   long test_dspal_get_version_info();
+   //long test_dspal_get_version_info();
    long test_clockid();
```
Comment (mcharleb) [inline test/dspal_tester/dspal_tester.idl:41]: Why no test of version API?

File: test/include/dspal_log.h [hunk 22]
```diff
@@ -43,7 +43,7 @@
 
 #define LOG_INFO(...) FARF(ALWAYS, __VA_ARGS__);
 #define LOG_ERR(...) FARF(ALWAYS, __VA_ARGS__);
-#define LOG_DEBUG(...) FARF(MEDIUM, __VA_ARGS__);
+#define LOG_DEBUG(...) FARF(ALWAYS, __VA_ARGS__);
```
Comment (mcharleb) [inline test/include/dspal_log.h:46]: DEBUG cannot be ALWAYS by default

File: test/push_tester.cmd [hunk 23]
```diff
@@ -0,0 +1,57 @@
...
+rem call start_dsp.cmd 3
+rem sleep 5
+rem
+rem adb push p:\poky\build\tmp\work\armv7a-vfp-neon-pokymllib32-linux-gnueabi\lib32-adsprpc\0.1-r0\image\usr\bin\sdsprpcd /usr/bin/.
+rem adb push P:\poky\build\tmp\sysroots\lib32-qcom-bsp\usr\lib\libadsprpc.so /usr/lib/.
+rem adb push P:\poky\build\tmp\sysroots\lib32-qcom-bsp\usr\lib\libsdsprpc.so /usr/lib/.
+rem adb shell chmod 755 /usr/lib/libadsprpc.so
+rem adb shell chmod 755 /usr/lib/libsdsprpc.so
+rem adb shell chmod 755 /usr/bin/sdsprpcd
+rem adb shell "/usr/bin/adsprpcd&"
```
Comment (mcharleb) [inline test/push_tester.cmd:16]: Should this not be started as a system service?

**Comment by mcharleb** [comment]
@wangxdflight Should this PR be closed? Is it obsoleted by your other PR?

**Comment by wangxdflight** [comment]
it is obsoleted. I created it just for your information. We have divided
the task and using a JIRA epic to track the tasks.

Thanks.
Larry

**Comment by mcharleb** [comment]
@wangxdflight Please rebase, and squash if you still want this merged

**Pull request closed without merging**